cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/MetadataSubscriberService.java

```diff
@@ -379,7 +381,14 @@ public void processMessage(MetadataMessage message, StructuredTableContext conte
       ProgramRunId programRunId = (ProgramRunId) message.getEntityId();
       FieldLineageInfo info;
       try {
-        info = message.getPayload(GSON, FieldLineageInfo.class);
+        Gson gson = new GsonBuilder()
+          .registerTypeAdapter(EntityId.class, new EntityIdTypeAdapter())
+          .registerTypeAdapter(MetadataOperation.class, new MetadataOperationTypeAdapter())
+          .registerTypeAdapter(Operation.class, new OperationTypeAdapter())
+          .registerTypeAdapter(EndPointField.class, new EndpointFieldDeserializer())
```

Comment:
Why can't we just add EndpointFieldDeserializer to exiting static field `GSON` in line 107,  instead of creating a new Gson object ?